dvc/scm/backend/base.py

```diff
@@ -0,0 +1,106 @@
+import os
+from typing import TYPE_CHECKING, Iterable, Optional, Tuple
+
+from dvc.scm.base import SCMError
+
+if TYPE_CHECKING:
+    from dvc.scm.git import Git
+
+
+class NoGitBackendError(SCMError):
+    def __init__(self, func):
+        super().__init__(f"No valid Git backend for '{func}'")
+
+
+class BaseGitBackend:
```

Comment:
Let's use abc right away
```suggestion
class BaseGitBackend(ABC):
```